pkg/ext-proc/handlers/server.go

```diff
@@ -37,7 +38,7 @@ type Server struct {
 }
 
 type Scheduler interface {
-	Schedule(b *scheduling.LLMRequest) (targetPod backend.Pod, err error)
+	Schedule(logger logr.Logger, b *scheduling.LLMRequest) (targetPod backend.Pod, err error)
```

Comment:
Here it is particularly ugly to pass a logger and not a context, but it seemed easier. Open to change.

Reply:
why not pass a context? This is a called by the request handler, so the context is there already.